agent.go

```diff
@@ -261,7 +276,14 @@ func (agent *agentS) fullRequestResponse(ctx context.Context, url string, method
 
 	if err == nil {
 		if j != nil {
-			req, err = http.NewRequest(method, url, bytes.NewBuffer(j))
+			b := bytes.NewBuffer(j)
+			if b.Len() > maxContentLength {
+				sensor.logger.Warn(`A batch of spans have been rejected because they are too large to be sent to the agent.`)
```

Comment:
*remark:* In Node we go the extra mile and inspect the batch of spans to check for the largest span in the batch, and then logging it in some abbreviated form IIRC. This is very helpful for troubleshooting situations where this error occurs (often it is caused by a few spans with very large annotations). But AFAICT we don't have this option here because the data is already serialized/marshalled? Or does `data` contain the spans in some kind of structure that can still be inspected? Just curious.

Reply:
Hi @basti1302, 

Thank you for bringing up this topic. I have seen these log messages for the node tracer. I was not sure about how useful it will be here. Because to find the top 5 biggest serialized spans we have to serialize them one by one and then use heap/sort/something else. All of these might potentially consume a big amount of memory and create extra load on the CPU and log system because we do not know how huge is data, we now only that it is > 5MB. This is a reason why I did not put it there initially.

Nevertheless, if it helps to solve problems with debugging, I think it is ok to have it. Anyway, we serialized this data anyway before sending it, so in the worst case, it should more than ~doubled the memory we already use. Also as we log it only once and with a warning log level, it should not create unexpecting long-lasting issues.

I added this functionality (https://github.com/instana/go-sensor/pull/332/commits/68cd5db652a43abec696d8c70d414117a63a9312), please review it. 

Thank you in advance.

Reply:
Awesome! 👍